lib/mixins/properties-mixin.html

```diff
@@ -0,0 +1,212 @@
+    const base = Polymer.PropertiesChanged(superClass);
+
+    /**
+     * Mixes `moreProps` into `props` but upgrades shorthand type
+     * syntax to { type: Type}.
+     *
+     * @param {Object} props Properties to normalize
+     * @return {Object} Copy of input `props` with normalized properties that
+     * are in the form {type: Type}
+     * @private
+     */
+    function normalizeProperties(props) {
+      const output = {};
+      for (let p in props) {
+        let o = props[p];
+        if (typeof o == 'function') {
+          o = { type: o };
+        }
+        output[p] = o;
+      }
+      return output;
+    }
+
+    /**
+     * Returns the super class constructor for the given class, if it is an
+     * instance of the PropertiesMixin.
+     *
+     * @param {PropertiesMixinConstructor} constructor PropertiesMixin constructor
+     * @return {PropertiesMixinConstructor} Super class constructor
+     */
+    function superForClass(constructor) {
+      const proto = /** @type {PropertiesMixinConstructor} */ (constructor).prototype;
+      const superCtor = Object.getPrototypeOf(proto).constructor;
+      if (superCtor.prototype instanceof PropertiesMixin) {
+        return superCtor;
+      }
+    }
+
+    /**
+     * Returns a memoized version of the `properties` object for the
+     * given class. Properties not in object format are converted to at
+     * least {type}.
+     *
+     * @param {PropertiesMixinConstructor} constructor PropertiesMixin constructor
+     * @return {Object} Memoized properties object
+     */
+    function ownProperties(constructor) {
+      if (!constructor.hasOwnProperty(
+        JSCompiler_renameProperty('__ownProperties', constructor))) {
+        const props = constructor.properties;
+        constructor.__ownProperties = props ? normalizeProperties(props) : null;
+      }
+      return constructor.__ownProperties;
+    }
+
+    /**
+     * @polymer
+     * @mixinClass
+     * @extends {base}
+     * @implements {Polymer_PropertiesMixin}
+     * @unrestricted
+     */
+    class PropertiesMixin extends base {
+
+      /**
+       * Implements standard custom elements getter to observes the attributes
+       * listed in `properties`.
+       */
+      static get observedAttributes() {
+        const props = this._properties;
+        return props ? Object.keys(props).map(p => {
+          return this.attributeNameForProperty(p);
+        }) : [];
+      }
+
+      /**
+       * Finalizes an element definition, including ensuring any super classes
+       * are also finalized. This includes ensuring property
+       * accessors exist on the element prototype. This method calls
+       * `_finalizeClass` to finalize each constructor in the prototype chain.
+       */
+       static finalize() {
+        if (!this.hasOwnProperty(JSCompiler_renameProperty('__finalized', this))) {
+          const superCtor = superForClass(this);
+          if (superCtor) {
+            superCtor.finalize();
```

Comment:
Justin idea: eliminate `superForClass` and instead:
```js
if (super.prototype instanceof PropertiesMixin) {
   Object.getPrototypeOf(this).finalize();
}
```
Or, can we not do this because closure doesn't support property access off of super?

Reply:
That's right and there's another spot we'd need the same check. Will rename to `superPropertiesClass`